phpBB/includes/functions_posting.php

```diff
@@ -2507,7 +2507,7 @@ function submit_post($mode, $subject, $username, $topic_type, &$poll_ary, &$data
 	}
 
 	$url = (!$params) ? "{$phpbb_root_path}viewforum.$phpEx" : "{$phpbb_root_path}viewtopic.$phpEx";
-	$url = append_sid($url, 'f=' . $data_ary['forum_id'] . $params) . $add_anchor;
+	$url = append_sid($url, $params) . $add_anchor;
```

Comment:
With the now missing `f=` parameter, the url will be like `viewtopic.php?&t=1&p=83#p83`

Reply:
True. And with no params, the URL would be `viewforum.php?#83`

This should do it to replace L2493-L2513 with:
```
	$params = $add_anchor = '';
	$url = "{$phpbb_root_path}viewtopic.$phpEx";

	if ($post_visibility == ITEM_APPROVED ||
		($auth->acl_get('m_softdelete', $data_ary['forum_id']) && $post_visibility == ITEM_DELETED) ||
		($auth->acl_get('m_approve', $data_ary['forum_id']) && in_array($post_visibility, array(ITEM_UNAPPROVED, ITEM_REAPPROVE))))
	{
		if ($mode != 'post')
		{
			$params = 'p=' . $data_ary['post_id'];
			$add_anchor = '#p' . $data_ary['post_id'];
		}
		else
		{
			$params = 't=' . $data_ary['topic_id'];
		}
	}
	else if ($mode != 'post' && $post_mode != 'edit_first_post' && $post_mode != 'edit_topic')
	{
		$params = 't=' . $data_ary['topic_id'];
	}
	else
	{
		$url = "{$phpbb_root_path}viewforum.$phpEx"
		$params = 'f=' . $data_ary['forum_id'];
	}
	$url = append_sid($url, $params) . $add_anchor;
```
This even simplifies the URL more consequently: https://github.com/phpbb/phpbb/blob/2df7aa3082166b3df945ba6a8757147953d378a2/phpBB/includes/functions_posting.php#L2493-L2513
- If any if the if statements matched, `$params` was assigned, in which case it became a `viewtopic.php` URL, so does not required the forum ID. So I added an `else` statement to use the `viewtopic.php` URL and add the forum ID, while it by default uses the `viewtopic.php` and does not add the forum ID.
- If a post was selected, the post ID and anchor was added, which again makes the topic ID obsolete. So I added an `else` statement to add the topic ID only if the post ID is not added.
- Since now only a single ID is added in every case, making the ampersand obsolete.

Reply:
I have actually now just changed this to using an array for `$params`

Reply:
`append_sid` handles an input array correctly automatically?

The issue stay that:
- topic ID + post ID are both set when the first both conditions (L2547 + L2553) are met, while only post ID is required.
- If none of the both parent conditions (L2547 + L2559) are met, the forum ID would be required (?) but is not set. If this cannot happen, then replacing the "if else" (L2559) condition with an "else" would make that clear and won't allow to pass no parameter in case of error.

Reply:
Should be resolved now